### PR TITLE
In LAMB optimizer, rename weight_decay_rate to weight_decay

### LINKED ISSUE #2610: LAMB Optimizer - rename weight_decay_rate to weight_decay
**Describe the bug**

Unlike the rest of the optimizers API, in LAMB the name of the argument for `weight_decay` is `weight_decay_rate`

In my experiments where I try various different optimizers, this makes me add a special condition for LAMB because of this different name of the argument.

**Code to reproduce the issue**

https://github.com/tensorflow/addons/blob/8cec33fcaaf1cf90aec7bdd55a0fcdbb251ce5c2/tensorflow_addons/optimizers/lamb.py#L44

### PR DESCRIPTION
**Pull request by ksachdeva**
# Description

Brief Description of the PR:

Fixes #2610 

## Type of change

- [x ] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [x] This PR addresses an already submitted issue for TensorFlow Addons
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

By writing a unit test

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*  Ran pytest

### CODE REVIEW

**Comment by bhack** [comment]
Tests are failing

**Comment by ksachdeva** [comment]
The tests are passing but seems to be an issue with wheel building on Ubuntu.

File: tensorflow_addons/optimizers/lamb.py [hunk 1]
```diff
@@ -74,6 +76,15 @@ def __init__(
               decay of learning rate. `lr` is included for backward
               compatibility, recommended to use `learning_rate` instead.
         """
+
+        if "weight_decay_rate" in kwargs:
+            warnings.warn(
+                "weight_decay_rate has been renamed to weight_decay",
```
Comment (bhack) [inline tensorflow_addons/optimizers/lamb.py:82]: Palese claim an Addons version as in the PR example I've mentioned

**Review by bhack — APPROVED** [review]

File: tensorflow_addons/optimizers/lamb.py (hunk 1 above)
Comment (ksachdeva) [reply tensorflow_addons/optimizers/lamb.py]: added

**Review by bhack — APPROVED** [review]
